.github/workflows/publish-categorical-packages.yml

```
@@ -0,0 +1,107 @@
+# Description: This workflow runs unit + e2e tests, then publishes UI packages
+#              to `@categorical-packages` NPM tag.
+#
+# Triggered by: merge to `categorical-packages/main`
+
+name: Test and Publish / categorical-packages
+
+concurrency:
+  group: categorical-packages-${{ github.sha }}
+
+on:
+  push:
+    branches: [categorical-packages/main]
+
+permissions:
+  id-token: write # This is required for requesting the JWT
+  contents: read # This is required for actions/checkout
+
+jobs:
+  setup-cache:
+    uses: aws-amplify/amplify-ui/.github/workflows/reusable-setup-cache.yml@main
```

Think this should be:
```suggestion
    uses: aws-amplify/amplify-ui/.github/workflows/reusable-setup-cache.yml@categorical-packages/main
```

cc @wlee221 for confirmation

---

Yep, let's target this feature branch.

---

Thanks! @sreeramsama You will also want to make the update on lines 27 & 34

---

We don't really change the reusable-setup-cache workflow, so I don't think it would make any difference.

---

I have another PR for these changes. Should I just include them here too so it is easier to revert as well?
- https://github.com/aws-amplify/amplify-ui/pull/3897

---

That's fair. We should at least change unit and e2e ones though. And while we're at that, i think it makes sense to be just change them all to be consistent.

---

Yes please. 

Regarding:
> We don't really change the reusable-setup-cache workflow, so I don't think it would make any difference.

Of the opinion that having consistency makes for easier understanding in for future readers, regardless os whether the code is temporary

---

Sure, will change all for consistency; and will update this PR to include changes from [ci: run reusable workflows in categorical-packages/main branch](https://github.com/aws-amplify/amplify-ui/pull/3897)